### PR TITLE
Prohibit non zero versions for dev sources

### LINKED ISSUE #854: Implement `+version 0.0.0` check for eo modules
The version must not be changed manually during development and must always be `0.0.0`.

Maven verifier plugin can be used to check presence of correct version.
https://maven.apache.org/plugins/maven-verifier-plugin/index.html

### PR DESCRIPTION
**Pull request by mximp**
closes #854

### CODE REVIEW

**Comment by mximp** [comment]
@volodya-lombrozo please review

**Review by volodya-lombrozo — CHANGES_REQUESTED** [review]
@mximp 

1. Maybe we can implement that check using `xlst`? I mean we can add just one more check to `eo-parser/src/main/resources/org/eolang/parser/errors`.

2. The tests is hard to read because they are using `xml` concatenation. Maybe we can use something like [this](https://github.com/objectionary/eo/blob/master/eo-parser/src/test/java/org/eolang/parser/PacksTest.java)?

**Comment by mximp** [comment]
@volodya-lombrozo Thanks for the hint! Made it with XSLT.
Please recheck

File: eo-parser/src/main/resources/org/eolang/parser/errors/zero-version.xsl [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+-->
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:eo="https://www.eolang.org" xmlns:xs="http://www.w3.org/2001/XMLSchema" id="zero-version" version="2.0">
+  <xsl:import href="/org/eolang/parser/_funcs.xsl"/>
+  <xsl:output encoding="UTF-8" method="xml"/>
+  <xsl:template match="/program/errors">
+    <xsl:copy>
+      <xsl:apply-templates select="node()|@*"/>
+      <xsl:for-each select="/program/metas/meta[head='version']">
+        <xsl:variable name="src" select="/program/@source"/>
+        <xsl:if test="matches($src, 'src/(?:main|test)') and not(./tail/text() = '0.0.0')">
```
Comment (volodya-lombrozo) [inline eo-parser/src/main/resources/org/eolang/parser/errors/zero-version.xsl:33]: @mximp The only thing which isn't clear it's why exactly `main|test`? 

I've found your comment: 

> The policy is not to update the version manually.
It should always be 0.0.0 in master and development branches.
It's incremented automatically during deployment process.

Why then it isn't a `master|test` ?

**Review by volodya-lombrozo — CHANGES_REQUESTED** [review]

**Review by volodya-lombrozo — CHANGES_REQUESTED** [review]
@mximp I got it, thank you. Everything looks good to me, but can I ask you to rename tests, please:
```
zero-version-correct-version.yml -> zero-version-for-main-correct.yml
zero-version-for-test.yml -> zero-version-for-test-wrong.yaml
zero-version-no-version.yml -> good
zero-version-no-version-not-applicable-source.yml -> good
zero-version.yml -> zero-version-for-main-wrong.yml
```
I was trying to understand which cases you considered and it take some time to understand.

**Comment by mximp** [comment]
@volodya-lombrozo please check now

**Review by volodya-lombrozo — APPROVED** [review]
@mximp Thanks. Looks good to me.
@yegor256 Could you merge that changes, please?

**Comment by mximp** [comment]
@yegor256 please merge the PR

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**